### PR TITLE
Replace `del` with `rimraf` package

### PR DESCRIPTION
**Pull request by dav-is**
Replaces use of `del` in favor of `rimraf` package because it's smaller and less complex
Removes `packages/next/lib/promisify.js` in favor of native promisify because it's leftover Node 6 polyfill.
Converts 2 files to TypeScript

### CODE REVIEW

File: packages/next/export/index.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ export default async function (dir, options, configuration) {
 
   // Initialize the output directory
   const outDir = options.outdir
-  await del(join(outDir, '*'))
+  await rimraf(join(outDir, '*'), (e) => { throw e })
```
Comment (wtgtybhertgeghgtwtg) [inline packages/next/export/index.js]: This no longer returns a Promise, so it can't be await'd.

Comment (dav-is) [reply packages/next/export/index.js]: Fixed :+1:

**Comment by wtgtybhertgeghgtwtg** [comment]
Is there a reason for this change?  It's not any simpler, since it has to be promisified, and `del` is still in the dependency tree via `autodll-webpack-plugin` and `recursive-copy`.

**Comment by timneutkens** [comment]
> Is there a reason for this change?

Actually it is simpler `del` exposes a async / sync api, uses globby and rimraf, the path to execution in this case is shorter and less code is loaded in eg `next export`

**Comment by wtgtybhertgeghgtwtg** [comment]
I agree `del` is much more bulky than it needs to be, but you are using the async API.  And `del` will be loaded in `next export` anyway, since you're using `recursive-copy`.

File: packages/next/server/hot-reloader.js [hunk 2]
```diff
@@ -161,7 +161,7 @@ export default class HotReloader {
   }
 
   async clean () {
-    return del(join(this.dir, this.config.distDir), { force: true })
+    return promisify(rimraf)(join(this.dir, this.config.distDir), { force: true })
```
Comment (wtgtybhertgeghgtwtg) [inline packages/next/server/hot-reloader.js]: This'll promisify it on every clean.  Want to hoist it?

**Comment by timneutkens** [comment]
@wtgtybhertgeghgtwtg we're going to drop recursive-copy or use only parts of it too, it does way more than we need too 💯

File: packages/next/server/hot-reloader.js (hunk 2 above)
Comment (timneutkens) [reply packages/next/server/hot-reloader.js]: Agreed it should be initialized right after the imports.

**Comment by wtgtybhertgeghgtwtg** [comment]
That seems reasonable.  There is a chance that `del` will become less massive and potentially more performant than `rimraf` (when `micromatch@4` drops and if `fast-glob` implements `hasMagic`).  Would it be possible to document the reason behind these changes so that can be explored later?

**Comment by timneutkens** [comment]
We can even drop the globbing, we don't use the globbing part, we just want to delete 1 directory in this case 😅

**Comment by wtgtybhertgeghgtwtg** [comment]
Oh, wow, I should have noticed that.

**Comment by timneutkens** [comment]
@wtgtybhertgeghgtwtg this is one of the reasons why I'm still not convinced moving to rimraf is even the correct thing to do, it could be even more performant to use fs.unlink directly in this case. But this is a good first step 👍

**Comment by timneutkens** [comment]
On a similar note we have the `glob` dep just to read `pages/**/*`, it could easily be switched to a readdir version 🤔

**Comment by timneutkens** [comment]
@wtgtybhertgeghgtwtg could you reach out to me on https://spectrum.chat/users/timneutkens / twitter.com/timneutkens?

**Review by timneutkens — APPROVED** [review]

**Pull request merged**